src/analytics/analytics.php

```diff
@@ -105,6 +102,27 @@ private static function is_wpvip_site() {
 			&& defined( 'FILES_CLIENT_SITE_ID' )
 			&& function_exists( '\Automattic\VIP\Stats\send_pixel' );
 	}
+
+	/**
+	 * Check if the current request should be sampled for analytics.
+	 * 
+	 * Current sampling algorithm is that every 10s, we send analytics.
+	 *
+	 * @return boolean true, if it should be sampled or false otherwise.
+	 */
+	private static function is_it_sampling_time() {
+		$current_timestamp = current_datetime();
+
+		// Get the seconds from the date.
+		$seconds = intval( $current_timestamp->format( 's' ) );
+
+		// Only send analytics every 10 seconds.
+		if ( 0 === ( $seconds % WPCOMVIP__BLOCK_DATA_API__STAT_SAMPLING_RATE_SEC ) ) {
```

Comment:
Someday we may want a hook so that customers can sent their own sampling rate no lower than X, but for a different day.

Reply:
I do see the value of that, especially if they are at a scale where even the tiniest change could have a large impact. We would need to figure out a good floor that we can use, that would still make the data useful to us. I'll add this to the backlog.